client/src/components/FilterBox/FilterBox.js

```diff
@@ -13,58 +16,133 @@ import {
+
+  const handleTag = (label, option) => (e) => {
+    e.preventDefault();
+    const options = selectedFilters[label] || [];
+    const optionIdx = options.indexOf(option);
+    let newOptions;
+    if (optionIdx === -1) {
+      newOptions = [...options, option];
+    } else {
+      options.splice(optionIdx, 1);
+      newOptions = options;
+    }
+    setSelectedFilters({ ...selectedFilters, [label]: newOptions });
+  };
+
+  const renderFilterOptions = (filters) => {
+    return filters.map((filter, idx) => (
+      <FilterOptionButton
+        handleClick={openModal(idx)}
+        key={idx}
+        label={filter.label}
+      />
+    ));
+  };
+
+  const renderFilterPanels = (filters) => {
+    return filters.map((filter, idx) => {
+      if (filter.label !== "Location") {
+        return (
+          <FilterAccordionPanel header={filter.label} key={idx}>
+            {Object.values(filter.options).map((option, idx) => (
+              <FilterTag
+                handleClick={handleTag(filter.label, option)}
+                label={option}
+                key={idx}
+                selected={
+                  selectedFilters[filter.label] &&
+                  selectedFilters[filter.label].includes(option)
+                }
+              />
+            ))}
+          </FilterAccordionPanel>
+        );
+      } else {
+        return (
+          <FilterAccordionPanel header={filter.label} key={idx}>
+            <LocationSearch
+              location={location}
+              handleLocation={handleLocation}
+              shareMyLocation={shareMyLocation}
+            />
+          </FilterAccordionPanel>
+        );
+      }
+    });
+  };
+
   return (
     <FilterBoxWrapper>
       <FilterTitle>Filter by</FilterTitle>
-      {Object.values(filterOptions).map((filter, idx) => (
-        <FilterOptionButton
-          handleClick={openModal(filter.label)}
-          key={idx}
-          label={filter.label}
-          icon={true}
-        />
-      ))}
+      {renderFilterOptions(filters)}
       <Modal
         popup
         visible={modal}
         onClose={closeModal}
         animationType="slide-up"
-        afterClose={() => {
-          console.log("afterClose");
-        }}
       >
-        <FilterAccordion className="my-accordion" onChange={onChange}>
-          {Object.values(filterOptions).map((filter) => (
-            <FilterAccordionPanel header={filter.label}>
-              {Object.values(filter.options).map((option) => (
-                <FilterTag label={option} />
-              ))}
-            </FilterAccordionPanel>
-          ))}
+        <FilterAccordion
+          defaultActiveKey={activePanel}
+          className="my-accordion"
+        >
+          {renderFilterPanels(filters)}
         </FilterAccordion>
+        <CustomList center="true">
+          <List.Item>
+            <CustomButton
+              // for some reason react wants boolean values for styled components
```

Comment:
hmm good point, was just going to comment :) can possibly be fixed somehow but not that important